plugins/pencilblue/controllers/actions/login.js

```diff
@@ -16,67 +16,67 @@
 */
 'use strict';
 
+const passport = require('koa-passport');
+
 module.exports = function LoginActionControllerModule(pb) {
 
     //dependencies
-    var util               = pb.util;
     var FormController     = pb.FormController;
-    var FormAuthentication = pb.FormAuthentication;
 
     /**
      * Authenticates a user
      * @class LoginActionController
      * @constructor
      * @extends FormController
      */
-    function LoginActionController(){}
-    util.inherits(LoginActionController, FormController);
-
-    /**
-     *
-     * @method onPostParamsRetrieved
-     * @param {Object} post
-     * @param {Function} cb
-     */
-    LoginActionController.prototype.render = function (cb) {
+    class LoginActionController extends pb.BaseController {
+        render (cb) {
             this.sanitizeObject(this.body);
-            this.onPostParamsRetrieved(this.body, cb);
-    };
-    LoginActionController.prototype.onPostParamsRetrieved = function(post, cb) {
-        var self         = this;
-        var adminAttempt = this.query.admin_attempt ? true : false;
-
-        var options = post;
-        options.access_level = adminAttempt ? pb.SecurityService.ACCESS_WRITER : pb.SecurityService.ACCESS_USER;
-        options.site = self.site;
-        pb.security.authenticateSession(this.session, options, new FormAuthentication(), function(err, user) {
-            if(util.isError(err) || user === null)  {
-                self.loginError(adminAttempt, cb);
-                return;
+            this._setupLoginContext();
+            this._doLogin(cb);
+        }
+        loginError (cb) {
+            this.session.error = this.ls.g('login.INVALID_LOGIN');
+            if(this.isAdminLogin){
+                return this.redirect('/admin/login', cb);
             }
 
-            //redirect
-            var location = '/';
-            if (self.session.on_login !== undefined) {
-                location = self.session.on_login;
-                delete self.session.on_login;
-            }
-            else if(adminAttempt) {
-                location = '/admin';
-            }
-            self.redirect(location, cb);
-        });
-    };
+            return this.redirect('/user/login', cb);
+        };
 
-    LoginActionController.prototype.loginError = function(adminAttempt, cb) {
-        this.session.error = this.ls.g('login.INVALID_LOGIN');
-        if(adminAttempt){
-            this.redirect('/admin/login', cb);
-            return;
+
+        _doLogin (cb) {
+            let redirectLocation = this.redirectLink;
+
+            return passport.authenticate('custom-local', (err, user) => {
+                if (!user) {
+                    return this.loginError(this.isAdminLogin, cb);
+                }
+                this.redirect(redirectLocation, cb);
```

Comment:
is this going to perform the redirection ? or you need to redirect through ctx I'm not sure about that